src/main/java/dev/railroadide/railroad/java/JDKManager.java

```diff
@@ -0,0 +1,178 @@
+package dev.railroadide.railroad.java;
+
+import dev.railroadide.core.utility.OperatingSystem;
+import dev.railroadide.railroad.utility.JavaVersion;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class JDKManager {
+
+    private static final List<String> WIN_JDK_PATHS = List.of(
+        "C:\\Program Files\\Java",
+        "C:\\Program Files (x86)\\Java",
+        "C:\\Program Files\\Eclipse Adoptium"
+    );
+    private static final List<String> MAC_JDK_PATHS = List.of(
+        "/Library/Java/JavaVirtualMachines"
+    );
+    private static final List<String> LINUX_JDK_PATHS = List.of(
+        "/usr/lib/jvm",
+        "/usr/java"
+    );
+    private static final String JAVA_EXECUTABLE = getJavaExecutable();
+
+    private static List<JDK> jdks;
+
+    public static void init() {
+        jdks = discoverJDKs();
+    }
```

Comment:
I'm not a massive fan of this. I guess it works, it just doesn't feel like this is the most optimal way to achieve it. My personal recommendation, at least to start with, is:

```suggestion
    private static final List<JDK> jdks = new CopyOnWriteArrayList<>();

    public static void init() {
        jdks.addAll(discoverJDKs());
    }
```

Then afterwards, its worth considering a way for a user to define custom paths in which JDKs can live, or even directly add/remove a specific JDK. This is because otherwise this system is strictly limited to those locations.

It is also worth noting another common location, is `user.home/.jdks/`, as this is where IntelliJ often installs JDKs to.